internal/provider/resource_ko_image.go

```diff
@@ -178,6 +187,23 @@ func (o *buildOptions) makeBuilder(ctx context.Context) (*build.Caching, error)
 var baseImages sync.Map // Cache of base image lookups.
 
 func doBuild(ctx context.Context, opts buildOptions) (string, error) {
+	if opts.koDockerRepo == "" && opts.imageRepo == "" {
+		return "", errors.New("one of KO_DOCKER_REPO env var, or provider `docker_repo` or `repo`, or image resource `repo` must be set")
+	}
+	po := []publish.Option{publish.WithAuthFromKeychain(authn.DefaultKeychain)}
```

Comment:
rando thought: should we make this use the helpers automatically?

I think we did it in `ko` itself, so it might be nice for parity.

Reply:
Yeah good call. I'll send a separate PR.